.github/workflows/pr_validation.yml

```diff
@@ -24,10 +24,10 @@ jobs:
           tests:
             - 'tests/**'
 
-    - name: Set up Python 3.10
+    - name: Set up Python 3.9
```

Comment:
Why the downgrade?

Reply:
Somehow this caused GitHub to look for Python 3.1, which it didn't find. Probably something on the GitHub side. Anyways, it shouldn't be a problem to test it with 3.9 because of backward compatibility